app/services/service_provider_request_handler.rb

```diff
@@ -22,7 +22,9 @@ def call
   attr_reader :url, :session, :protocol_request, :protocol
 
   def ial
-    Saml::Idp::Constants::IAL2_AUTHN_CONTEXTS.include?(protocol.ial) ? 2 : 1
+    uri = URI.parse(protocol.ial)
```

Comment:
leaving a comment here to remind myself to write a test for this behavior. it is necessary for IAL2 strict